### ISSUE TITLE
darchtio-srf types are not correctly exported using TS

### ISSUE THREAD

**Issue by hadar-laguna** [drachtio/drachtio-srf#181]
in the update to 4.5.40 you added the namespace Srf and `export = Srf.Srf;`
Which caused all of the other types (Dialog, SipResponse, SipRequest, CreateUACOptions and many more)
to not be accessible.

Please fix this.
Thanks!

**Comment by davehorton** [comment]
can you refer to the problematic commit that you see?

**Comment by danvirsen** [comment]
Not OP, but I was in the process of updating drachtio-srf when I noticed this and found that there was an issue created only hours ago.

This is the PR: https://github.com/drachtio/drachtio-srf/pull/176

**Comment by hadar-laguna** [comment]
Also, it seems the function "options" is missing from types (srf.options( ...))

**Comment by hadar-laguna** [comment]
You should change it to 
```ts
    export = Srf;
```

and add options function inside the class Srf.

**Comment by danvirsen** [comment]
I created a new branch in my fork for this.
Could you please help me test?
If something is missing, feel free to make changes yourself. Most methods have multiple overloads so it's completely possible

**Comment by danvirsen** [comment]
I created a new branch in my fork for this.
Could you please help me test? You should be able to just change the path in 
If something is missing, feel free to make changes yourself.
A lot of methods have multiple overloads so it's completely possible that a couple are still missing.

I haven't checked out or added the "options" method yet.

https://github.com/danvirsen/drachtio-srf/tree/bugfix/181

**Comment by danvirsen** [comment]
I have added all missing (I hope) middleware methods for "srf".

**Comment by hadar-laguna** [comment]
Hey, So i checked it, and its still problematic since the typing is kind of "lying" as to the whole class.
    export = Srf;
is not necessarily the solution, i don't have a lot of time, i will try to fix it later this week and update you, or someone else can try it!

**Comment by danvirsen** [comment]
Yes, that's true. I did the changes blind but when I tested my changes I did get some "implicit any type" errors and other warnings that some types weren't available. So things weren't exposed correctly.
I have adressed that now (I think) and pushed the changes, and my application is at least happy again.

The codebase is JavaSCript and since the types are managed manually they will always "lie" I suppose. They're not connected to the actual code in any way which make some things that aren't documented lag behind and only surface when someone wants to use a specific call or function/method.

It's a risk to have manually handled types in the repository. Intellisense will only show what the types contain, which like I said aren't actually connected to the code, so there's still the risk of runtime errors due to the types being wrong.

I'm not sure if there is a good way to try and automate this.

One solution might be to simply remove the types, but they do contain a couple of ondocumented ways to call certain methods so I do think they add some value.

**Comment by hadar-laguna** [comment]
looks good! confirmed working!

**Comment by hadar-laguna** [comment]
looks good!
working for me as such:

```ts
import * as Srf from 'drachtio-srf';
import { CreateUACOptions, Dialog, SrfRequest, SrfResponse } from 'drachtio-srf';

  // eslint-disable-next-line
  // @ts-ignore
const srf: Srf.default = new Srf();
```

its not the most typescripty but it works..

**Comment by danvirsen** [comment]
Since the default export export the class Srf, I use it like this:

```typescript
import Srf, { CreateUACOptions, Dialog, SrfRequest, SrfResponse } from "drachtio-srf";
const srf = new Srf();
```

**Comment by danvirsen** [comment]
I was thinking that it might make sense to move the types to the DefinitelyTyped project instead of having them in this repository.

https://github.com/DefinitelyTyped/DefinitelyTyped

I can give that a go on Monday.

**Comment by orgads** [comment]
Proposed fix:
- #185

**Comment by danvirsen** [comment]
I have added a comment to your PR but overall I think it looks good.
The bigger question still remains for me though: Should the types be in this repository at all?

I think the best thing would be to add drachtio-srf to DefinitelyTyped and remove the types from this repository. That way Dave and the other maintainers don't need to be bothered with TypeScript things (this is a JavaScript project after all), and fixes to the types themselves doesn't require getting a PR approved and waiting for a new release of the package.

Unfortunately I don't have time in the nearest future to create a valid test suite for drachtio-srf in the DefinitelyTyped project. I think that would be the best solution if anyone has the opportunity to do that.

**Comment by orgads** [comment]
@davehorton what do you prefer? Keep the types here or create a package in DefinitelyTyped?

**Comment by davehorton** [comment]
I would prefer to move to DefinitelyTyped.  I will merge the PR for now but really it would be nice if it could be in a separate project IMO
